sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -24,92 +29,125 @@ enum class architecture {
   //   - the unique ID of the new architecture in SYCL RT source code to support
   //     querying the device architecture
   //
-  x86_64,
-  intel_cpu_spr,
-  intel_cpu_gnr,
-  intel_gpu_bdw,
-  intel_gpu_skl,
-  intel_gpu_kbl,
-  intel_gpu_cfl,
-  intel_gpu_apl,
+  // Important note about keeping architecture IDs below unique:
+  //   - the architecture ID must be a hex number with 16 digits
+  //   - the architecture ID must suit the following template:
+  //     0x AA BBBBBBBBBBBB CC (without spaces), where
+  //       - AA is 2-digit ID of category which must be unique
```

Comment:
```suggestion
  //       - AA is 2-digit ID of the architecture family which must be unique
```

The terms "family" and "category" are different concepts in the spec.  Each architecture ID belongs to exactly one family (e.g. "Intel CPU family", "Nvidia GPU family", etc.).  There is also a category that matches each family (e.g, `arch_category::intel_gpu` matches the "Intel GPU family").  However, I expect to add more categories soon.

In general, an architecture ID can belong to many different categories.  For example, `architecture::intel_gpu_pvc` will belong to both categories `arch_category::intel_gpu` and `arch_category::intel_gpu_xe`.